### PR TITLE
Feature: Wishlist managment

### LINKED ISSUE #4: Epic: Wishlist management
# Epic Wishlist Management

Feature: Wishlist Management

## Epic Description

This Feature will add wishlist management in our e-comm app

## List of Tasks (Complete in order)

- [ ] Wishlist icon should take to wishlist page
- [ ] Remove from wishlist button
- [ ] Adding to cart button
- [ ] If the cart already contains that item, it's only increasing the quantity.

### PR DESCRIPTION
**Pull request by Pratham82**
# Wishlist management

[Epic: Wishlist management #4](https://github.com/Pratham82/e-comm-react/issues/4)

## Changes made

- [x] Created wishlist context, hook, and services
- [x] Added wishlist toggle feature in product component and product in cart component
- [x] Added wishlist product count in the navbar

## How does the solution address the problem

This PR will add wishlist management feature in app

Preview: https://e-comm-react-git-wishlist-managment-pratham82.vercel.app/

## Linked issues

Resolves #4

### CODE REVIEW

**Review by Balakumar3018 — APPROVED** [review]
Good to merge

File: src/pages/wishlist/index.tsx [hunk 1]
```diff
@@ -1,17 +1,52 @@
...
+              ),
+            )
+          : ""}
+        {!wishlistData.length && (
+          <div>
+            <div className="h6 mt-60">
+              Wishlist is empty
+              <span className="pl-6 link-tag">
+                <Link to="/products">Check products</Link>{" "}
+              </span>
```
Comment (arulk8) [inline src/pages/wishlist/index.tsx:45]: do we need this` { "" }`

File: src/services/wishlist/indext.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import staticToken from "assets/data";
+import axios from "axios";
+import { ADD_TO_WISHLIST, REMOVE_FROM_WISHLIST } from "types/cart";
+
+export const addToWishlist = async (product: any, wishlistDispatch: any) => {
+  try {
+    await axios.post(
+      "/api/user/wishlist",
```
Comment (arulk8) [inline src/services/wishlist/indext.ts]: can it be moved to `const WishListBaseURL = '/api/user/wishlist'`

**Review by arulk8 — APPROVED** [review]

File: src/pages/wishlist/index.tsx (hunk 1 above)
Comment (Pratham82) [reply src/pages/wishlist/index.tsx:45]: Added by prettier I'll remove this

File: src/services/wishlist/indext.ts (hunk 2 above)
Comment (Pratham82) [reply src/services/wishlist/indext.ts]: Yes I'll update that

**Pull request merged**